File: google/code-jam/2020/Q /4.py
test,b=[int(x) for x in input().split()]
t=test

def opp(s):
    x=""
    for i in s:
        if i=='0':
            x+='1'
        if i=='1':
            x+='0'
    return x

while t and b==10:
    t-=1
    s=''
    for i in range(10):
        print(i+1)
        s+=input()

    print(s)
    verdict=input()
    if verdict!="Y" :
        break

while t and b==20:
    t-=1
    left,mid,right='','',''
    s='1'*20
    for i in range(10):
        print(i+1)
        left+=input()

    for i in range(10):
        print(i+6)
        mid+=input()

    if left[6:]!=mid[:6] and opp(left[6:])!=mid[:6]:
        if mid[6:]==left[6:][::-1]:
            print(left)
        
    else:
        print(s)
    verdict=input()
    if verdict!="Y" :
        break
        

File: google/code-jam/2020/Q /4.cpp
#include <bits/stdc++.h>
using namespace std;
#define forn(i, n) for(int i = 0; i < n; ++i)
typedef pair<int, int> pi;

int main(){
	int t,test,b;
	string verdict;
	cin>>test>>b;
	t=test;
	while(t-- && b==10){
		string s="";
		forn(i,10)
			s+='1';
		forn(i,10){
			cout<<i+1<<"\n";
			cin>>s[i];
		}

		cout<<s<<"\n";
	    cin>>verdict;
	    if(verdict!="Y")
	    	break;
	}
	while(t-- && b==20){
		string left="",mid="",right="";
		forn(i,10){
			left+='1';
			mid+='1';
			right+='1';
		}
		forn(i,10){
			cout<<i+1<<"\n";
			cin>>left[i];
		}

		forn(i,10){
			cout<<i+6<<"\n";
			cin>>mid[i];
		}

		cout<<s<<"\n";
	    cin>>verdict;
	    if(verdict!="Y")
	    	break;
	}
	return 0;
}